### ISSUE TITLE
Allow user-friendly "copy to clipboard" of an event

### ISSUE THREAD

**Issue by tsimbalar** [datalust/seq-tickets#669]
When trying to communicate the contents of a log event to someone who has no access to the Seq Server, I pretty often end up selecting/copying the contents of an event and pasting it in an e-mail/chat room ... 

That turns out to not be a specially pleasant experience, specially regarding the exception :p 

- If I select the "whole unfolded event" (starting before the date, and going all the way to the end of the exception), the properties attached to the event are somehow readable, but 
    - the text of the actions is also included
    - the text of the tagged properties is also included, without spaces
    - in the exception part, every line in the stack strace is separated by an empty line, like so (pasted in Notepad++)

```plain

02 Mar 2018 13:07:52.236
DocumentsXXXXProductionXXXX Unexpected exception when retrieving user profile : Error while getting profile for user XXXX XXXX  in XXXX
Id ▾
Level (Error) ▾
Type (0xC436D23C) ▾
Retain ▾
Send to app ▾
Raw JSON
AppComponent
Documents
AppName
Hector
CurrentUserAffectationId
XXXX
CurrentUserAgentId
XXXX
Environment
Production
ErrorMessage
Error while getting profile for user XXXX XXXXX  in XXXX
HttpRequestId
be33cc7d-2a55-4439-bd9b-6c0bdaadf120
HttpRequestUrl
https://XXXX/Documents?OrgLCID=1036&UserLCID=1036&id={0118A625-D4C5-E611-80EA-00155DB6700E}&orgname=XXXX&type=2&typename=contact
MachineName
XXXX
SourceContext
XXXX.Tools.HandleUserProfileProviderExceptionExceptionFilter

XXXX.Tools.UserProfileProviderException: Error while getting profile for user XXXX XXXX XXXX  in XXXX ---> System.InvalidOperationException: The user authentication failed!

   at Microsoft.Xrm.Sdk.Client.ServiceProxy`1.AuthenticateCore()

   at Microsoft.Xrm.Tooling.Connector.CrmWebSvc.ManagedTokenOrganizationServiceProxy.ValidateAuthentication()

   at Microsoft.Xrm.Sdk.Client.ServiceContextInitializer`1.Initialize(ServiceProxy`1 proxy)

   at Microsoft.Xrm.Sdk.Client.OrganizationServiceProxy.ExecuteCore(OrganizationRequest request)

   at Microsoft.Xrm.Sdk.Client.OrganizationServiceContext.Execute(OrganizationRequest request)

   at Microsoft.Xrm.Sdk.Linq.QueryProvider.RetrieveEntityCollection(OrganizationRequest request, NavigationSource source)

   at Microsoft.Xrm.Sdk.Linq.QueryProvider.Execute(QueryExpression qe, Boolean throwIfSequenceIsEmpty, Boolean throwIfSequenceNotSingle, Projection projection, NavigationSource source, List`1 linkLookups, String& pagingCookie, Boolean& moreRecords)

   at Microsoft.Xrm.Sdk.Linq.QueryProvider.Execute[TElement](QueryExpression qe, Boolean throwIfSequenceIsEmpty, Boolean throwIfSequenceNotSingle, Projection projection, NavigationSource source, List`1 linkLookups)

   at Microsoft.Xrm.Sdk.Linq.QueryProvider.Execute[TElement](Expression expression)

   at Microsoft.Xrm.Sdk.Linq.QueryProvider.System.Linq.IQueryProvider.Execute[TResult](Expression expression)

   at XXXX.Data.Dynamics.Sdk.DynamicsSdkClient.GetUserByIdAffectation(Int32 idAffectation) in D:\agentv2-02\_work\91\s\XXXX.Data\Dynamics\Sdk\DynamicsSdkClient.cs:line 32

   at XXXX.Tools.HectorUserProfileProvider.<GetUserProfileInOrganization>d__4.MoveNext() in D:\agentv2-02\_work\91\s\Hector.Documents\Tools\HectorUserProfileProvider.cs:line 31

   --- End of inner exception stack trace ---

   at XXXX.Tools.HectorUserProfileProvider.<GetUserProfileInOrganization>d__4.MoveNext() in D:\agentv2-02\_work\91\s\Hector.Documents\Tools\HectorUserProfileProvider.cs:line 57

--- End of stack trace from previous location where exception was thrown ---

   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()

   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)

   at XXXX.Controllers.DocumentController.<ListDocuments>d__9.MoveNext() in D:\agentv2-02\_work\91\s\Hector.Documents\Controllers\DocumentController.cs:line 63

--- End of stack trace from previous location where exception was thrown ---

   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()

   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)

   at System.Web.Mvc.Async.TaskAsyncActionDescriptor.EndExecute(IAsyncResult asyncResult)

   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeAsynchronousActionMethod>b__36(IAsyncResult asyncResult)

   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)

   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3d()

   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f()

   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)

   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass21.<>c__DisplayClass2b.<BeginInvokeAction>b__1c()

   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass21.<BeginInvokeAction>b__1e(IAsyncResult asyncResult)
```

- If I try selecting only the exception (all the text in the red zone), I get the exception's stack trace, but no new lines at all, which is pretty hard to parse for a human being :)  : 

```plain
XXXX.Tools.UserProfileProviderException: Error while getting profile for user YYYY in ZZZZZ ---> System.InvalidOperationException: The user authentication failed! at Microsoft.Xrm.Sdk.Client.ServiceProxy`1.AuthenticateCore() at Microsoft.Xrm.Tooling.Connector.CrmWebSvc.ManagedTokenOrganizationServiceProxy.ValidateAuthentication() at Microsoft.Xrm.Sdk.Client.ServiceContextInitializer`1.Initialize(ServiceProxy`1 proxy) at Microsoft.Xrm.Sdk.Client.OrganizationServiceProxy.ExecuteCore(OrganizationRequest request) at Microsoft.Xrm.Sdk.Client.OrganizationServiceContext.Execute(OrganizationRequest request) at Microsoft.Xrm.Sdk.Linq.QueryProvider.RetrieveEntityCollection(OrganizationRequest request, NavigationSource source) at Microsoft.Xrm.Sdk.Linq.QueryProvider.Execute(QueryExpression qe, Boolean throwIfSequenceIsEmpty, Boolean throwIfSequenceNotSingle, Projection projection, NavigationSource source, List`1 linkLookups, String& pagingCookie, Boolean& moreRecords) at Microsoft.Xrm.Sdk.Linq.QueryProvider.Execute[TElement](QueryExpression qe, Boolean throwIfSequenceIsEmpty, Boolean throwIfSequenceNotSingle, Projection projection, NavigationSource source, List`1 linkLookups) at Microsoft.Xrm.Sdk.Linq.QueryProvider.Execute[TElement](Expression expression) at Microsoft.Xrm.Sdk.Linq.QueryProvider.System.Linq.IQueryProvider.Execute[TResult](Expression expression) at XXXX.Data.Dynamics.Sdk.DynamicsSdkClient.GetUserByIdAffectation(Int32 idAffectation) in D:\agentv2-02\_work\91\s\XXXX.Data\Dynamics\Sdk\DynamicsSdkClient.cs:line 32 at XXXX.Tools.ZZZZZUserProfileProvider.<GetUserProfileInOrganization>d__4.MoveNext() in D:\agentv2-02\_work\91\s\ZZZZZ.Documents\Tools\ZZZZZUserProfileProvider.cs:line 31 --- End of inner exception stack trace --- at XXXX.Tools.ZZZZZUserProfileProvider.<GetUserProfileInOrganization>d__4.MoveNext() in D:\agentv2-02\_work\91\s\ZZZZZ.Documents\Tools\ZZZZZUserProfileProvider.cs:line 57 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at XXXX.Controllers.DocumentController.<ListDocuments>d__9.MoveNext() in D:\agentv2-02\_work\91\s\ZZZZZ.Documents\Controllers\DocumentController.cs:line 63 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Web.Mvc.Async.TaskAsyncActionDescriptor.EndExecute(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass37.<BeginInvokeAsynchronousActionMethod>b__36(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3d() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f() at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass21.<>c__DisplayClass2b.<BeginInvokeAction>b__1c() at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass21.<BeginInvokeAction>b__1e(IAsyncResult asyncResult)
```

It would be great to have a mechanism to copy the contents of an event to the clipboard in a human-friendly format.

*(as of Seq 4.0.60.0)*

**Comment by nblumhardt** [comment]
Today's the day for clearing out things we'd love to do but can't schedule, and we're afraid this is another of them. We'll close it for now, since the "Raw JSON" download almost covers it, but we'll also keep it in mind in the future. Thanks @tsimbalar !
